### PR TITLE
Rewriting description to align with code and video

### PR DESCRIPTION
**Pull request by hollowcrust**
### Description

I rewrite the description of Floyd Cycle Detection Algorithm to align with the C++ and C code snippets and the video given, and give step-by-step illustration of how the algorithm works in the example.

Fixes #122

### Checklist

- [x] A description of the changes in this PR is mentioned above.
- [x] All the new and existing checks pass.
- [x] No plagiarized, duplicated, or repetitive documentation that has been directly copied from another source.
- [x] If it's a new explanation, it contains solid, understandable, and accessible information.
- [x] I have read the whole [contributing guidelines](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/CONTRIBUTING.md) and agree to the [Code of Conduct](https://github.com/TheAlgorithms/.github/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md [hunk 1]
```diff
@@ -5,34 +5,68 @@
 Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
 
 ## Approach
+- We can imagine the array `arr` as a directed graph where each element is a node. Node with value `arr[i]` will point to node with value `arr[arr[i]]`.
```
Comment (appgurueu) [inline en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md:8]: I find this confusing. I'd just say "`arr[i]` is the index of the node to which the i-th node points". Perhaps give a small example, like `[1, 0]`, and explain it.

File: en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md [hunk 2]
```diff
@@ -5,34 +5,68 @@
...
+- The problem now becomes finding the node where two nodes with same value point towards.
```
Comment (appgurueu) [inline en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md:9]: That sounds like something where two nodes point to the same node, which is not a cycle. A cycle is when there is a path from a node to itself. So for example in `[1, 0]`, you start at 0, then go to 1, then go back to zero.

File: en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
 
 ## Approach
-- We can imagine the array `arr` as a directed graph where each element is a node. Node with value `arr[i]` will point to node with value `arr[arr[i]]`.
-- The problem now becomes finding the node where two nodes with same value point towards.
-- Since a cycle is formed when two nodes with same value point towards the same node, we can use Floyd Cycle Detection Algorithm (also known as "Hare and Tortoise algorithm") to solve the problem.
-- The idea of algorithm is to maintain two pointers, `hare` and `tortoise` that iterate the array at different "speeds" (just like the fable). The details are as follow:
+- We can imagine the array `arr` as a directed graph where each element is a node. `arr[i]` is the index of the node to which the i-th node points.
+- For example, given the `arr = [1, 3, 4, 2, 3]`, we can represent `arr` as the following <br></br>
+![image](https://github.com/hollowcrust/Algorithms-Explanation/assets/72879387/a4afacb6-399b-44b6-96e9-b2fdfd75ae9d)
```
Comment (appgurueu) [inline en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md:10]: You should add the image in this PR and use a relative link.

Comment (hollowcrust) [reply en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md]: Hi I uploaded the images in a separate folder and put the relative links in the markdown.

**Review by appgurueu — APPROVED** [review]

**Comment by hollowcrust** [comment]
Thanks a lot!

**Comment by appgurueu** [comment]
> Thanks a lot!

Thank you for your PR :)

**Pull request merged**